rust/projects/tools/answers.md

```diff
@@ -5,19 +5,86 @@
 > What does this directory layout suggest about how `lib.rs`, `main.rs`, and
 `tests.rs` are compiled, and how they are linked to each other?
 
-Answer: TODO
+Answer: The compilation of our library file `lib.rs` is independent from
+`main.rs` and `tests.rs`. The library becomes an implicit dependency of
+`main.rs` and `tests.rs`, and is linked to the binaries `main.rs` and
+`tests.rs` compile into.
 
 ## Question B
 
 > In practice we might not name `src/bin/main.rs` as `main.rs`. Why not? What's
 the name of our binary? What are two ways we could change the name of that
 binary?
 
-Answer: TODO
+Answer: If you have tried `cargo run`, you should have seen this line:
+
+```
+     Running `target/debug/main`
+```
+
+It says the executable name we are running is `main`, which is identical to
+the name of the source file. In general, we expect our binary name to be
+meaningful instead of `main`.
+
+The simplest way to change the name is to just rename the source file.
+For example, rename `main.rs` to `kvs.rs` and recompile, you will find the
+binary name changes to `kvs` accordingly.
+
+You can also configure it in `Cargo.toml` by adding a `[[bin]]` section:
+
+```toml
+[[bin]]
+name = "kvs"
+path = "src/bin/main.rs"
+```
+
+In fact, if you move `src/bin/main.rs` to `src/main.rs`, the binary name
+will be the same as the package name (`kvs` in this case).
```

Comment:
Good answer. It makes me think we should just set up the project differently, with `main.rs` in `src/`, and get rid of the question. What do you think?

Reply:
Yes, I think we can just use `src/main.rs` but leave the question to be "modify the binary name in different ways".